libuavcan/include/uavcan/transport/can_acceptance_filter_configurator.hpp

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (C) 2014 Pavel Kirienko <pavel.kirienko@gmail.com>,
+ *                    Ilia  Sheremet <illia.sheremet@gmail.com>
+ */
+
+#ifndef CAN_ACCEPTANCE_FILTER_CONFIGURATOR_HPP_INCLUDED
+#define CAN_ACCEPTANCE_FILTER_CONFIGURATOR_HPP_INCLUDED
+
+#include <cassert>
+#include <uavcan/data_type.hpp>
+#include <uavcan/error.hpp>
+#include <uavcan/transport/dispatcher.hpp>
+#include <uavcan/node/abstract_node.hpp>
+#include <uavcan/build_config.hpp>
+#include <uavcan/util/multiset.hpp>
+
+namespace uavcan
+{
+
+class CanAcceptanceFilterConfigurator
+{
+    /**
+     * Below constants based on UAVCAN transport layer specification. Masks and ID's depends on message Priority,
+     * TypeID, TransferID (RequestNotResponse - for service types, BroadcastNotUnicast - for message types).
+     * For more details refer to uavcan.org/CAN_bus_transport_layer_specification.
+     * For clarity let's represent "i" as Data Type ID
+     * DefaultFilterMsgMask = 00111111111110000000000000000
+     * DefaultFilterMsgID   = 00iiiiiiiiiii0000000000000000, no need to explicitly define, since MsgID initialized as 0.
+     * DefaultFilterServiceRequestMask = 11111111111100000000000000000
+     * DefaultFilterServiceRequestID   = 101iiiiiiiii00000000000000000
+     * ServiceRespFrameMask = 11100000000000000000000000000
+     * ServiceRespFrameID   = 10000000000000000000000000000, all Service Response Frames are accepted by HW filters.
+     */
+    static const unsigned DefaultFilterMsgMask = 0x7FF0000;
+    static const unsigned DefaultFilterServiceRequestID = 0x14000000;
+    static const unsigned DefaultFilterServiceRequestMask = 0x1FFE0000;
+    static const unsigned ServiceRespFrameID = 0x10000000;
+    static const unsigned ServiceRespFrameMask = 0x1C000000;
+
+    typedef uavcan::Multiset<CanFilterConfig, 1> MultisetConfigArray;
+
+    static CanFilterConfig mergeFilters(CanFilterConfig &a_, CanFilterConfig &b_);
+    static uint8_t countBits(uint32_t n_);
+    uint16_t getNumFilters() const;
+
+    /**
+     * Fills the multiset_configs_ to proceed it with computeConfiguration()
+     */
+    int16_t fillArray();
```

Comment:
If it fills a multiset, why is it named `fillArray()`? :)

How about `loadInputConfiguration()` or whatever.